### PR TITLE
CVAT documentation v5

### LINKED ISSUE #586: Documentation: Add some examples of using search in a dashboard
Typing 'annotation' in search toolbar of dashboard does not filter tasks in 'annotation' mode.

### LINKED ISSUE #620: Documentation v0.5.0
Describe the following features in documentation:

- [ ] Remote data source (list of URLs to create an annotation task)
- [ ] Load/Dump annotations in different formats
- [ ] Linear interpolation for a single point
- [ ] Video frame filter
- [ ] Auto annotation (link on the cvat/apps/auto_annotation, a couple of examples)

### PR DESCRIPTION
**Pull request by TOsmanov**
Resolved #620
Resolved #586

### CODE REVIEW

**Comment by nmanovic** [comment]
@TOsmanov ,

1. cvat/apps/documentation/static/documentation/images/gif003.gif 
it is twice bigger and quality is similar or even worse. Could you please fix that?

2. cvat/apps/documentation/static/documentation/images/gif009_DETRAC.gif
~40Mb? It is taboo. The documentation will not be possible to open. Please compare all gif and be sure that they are the same size or smaller.

File: cvat/apps/documentation/user_guide.md [hunk 1]
```diff
@@ -169,18 +172,39 @@ Go to the [Django administration panel](http://localhost:8080/admin). There you
...
-    ![](static/documentation/images/image006.jpg)
+    ![](static/documentation/images/image006_DETRAC.jpg)
 
 1.  The Dashboard contains elements and each of them relates to a separate task. They are sorted in creation order.
     Each element contains: task name, preview, execution status, buttons, and one or more links.
     Each button is responsible for a specific function:
-    - ``Dump Annotation`` — download an annotation file from the task (xml format)
-    - ``Upload Annotation`` — upload an annotation file to the task (xml format)
+    - ``Dump Annotation`` — download an annotation file from a task. Several formats are available:
+      - [CVAT XML 1.1 for video](https://github.com/opencv/cvat/blob/develop/cvat/apps/documentation/xml_format.md#interpolation)
```
Comment (bsekachev) [inline cvat/apps/documentation/user_guide.md]: Let's use relative links everywhere. For example in this case: ``cvat/apps/documentation/xml_format.md#interpolation``

File: cvat/apps/documentation/user_guide.md [hunk 2]
```diff
@@ -169,18 +172,39 @@ Go to the [Django administration panel](http://localhost:8080/admin). There you
...
+      is highlighted if a task has the interpolation mode 
+      - [CVAT XML 1.1 for images](https://github.com/opencv/cvat/blob/develop/cvat/apps/documentation/xml_format.md#annotation)
```
Comment (bsekachev) [inline cvat/apps/documentation/user_guide.md]: ``cvat/apps/documentation/xml_format.md#annotation``

File: cvat/apps/documentation/user_guide.md [hunk 3]
```diff
@@ -169,18 +172,39 @@ Go to the [Django administration panel](http://localhost:8080/admin). There you
...
+      is highlighted if a task has the annotation mode 
+      - [PASCAL VOC ZIP 1.0](http://host.robots.ox.ac.uk/pascal/VOC/)
+      - [YOLO ZIP 1.0](https://pjreddie.com/darknet/yolo/)
+      - [COCO JSON 1.0](http://cocodataset.org/#format-data)
+      - ``MASK ZIP 1.0`` — archive contains a mask of each frame in the png format and a text file
+      with the value of each color
+      - [TFRecord ZIP 1.0](https://www.tensorflow.org/tutorials/load_data/tf_records)
+    - ``Upload Annotation`` — upload an annotation file to the task. Several formats are available:
+      - [CVAT XML 1.1](https://github.com/opencv/cvat/blob/develop/cvat/apps/documentation/xml_format.md)
```
Comment (bsekachev) [inline cvat/apps/documentation/user_guide.md]: ``cvat/apps/documentation/xml_format.md``

File: cvat/apps/documentation/user_guide.md [hunk 4]
```diff
@@ -216,15 +240,28 @@ You can "register" a model and "use" it after that to pre annotate your tasks.
 
 The model manager allows you to manage your deep learning (DL) models uploaded for auto annotation.
 Using the functionality you can upload, update or delete a specific DL model.
-Use "Auto annotation" button to pre annotate a task using one of your DL models. [Read more](/cvat/apps/auto_annotation)
+Use "Auto annotation" button to pre annotate a task using one of your DL models.
+[Read more](https://github.com/opencv/cvat/tree/develop/cvat/apps/auto_annotation)
```
Comment (bsekachev) [inline cvat/apps/documentation/user_guide.md]: ``cvat/apps/auto_annotation``

File: cvat/apps/documentation/user_guide.md [hunk 5]
```diff
@@ -216,15 +240,28 @@ You can "register" a model and "use" it after that to pre annotate your tasks.
...
 
 ![](static/documentation/images/image104.jpg)
 
 ### Search
 
 There are several options how to use the search.
-It may be task's name, mode (annotation or interpolation), owner (username), assignee (username),
-status (annotation, validation, completed), id.
+
+- Search in all fields (owner, assignee, task name, task status, task mode).
```
Comment (bsekachev) [inline cvat/apps/documentation/user_guide.md]: in -> within

File: cvat/apps/documentation/user_guide.md [hunk 6]
```diff
@@ -403,18 +440,31 @@ Usage examples:
...
-1.  The annotation will be written into **.xml** file. To find the annotation
-    file, you should go to the directory where your browser saves downloaded
-    files by default. For more information visit [xml format page](/cvat/apps/documentation/xml_format.md).
+1.  After that, press ``Dump Annotation`` button.
 
-    ![](static/documentation/images/image029.jpg)
+    ![](static/documentation/images/image118.jpg)
+
+1.  Choose format dump annotation file. Dump annotation are available in several formats:
+    - [CVAT XML 1.1 for video](https://github.com/opencv/cvat/blob/develop/cvat/apps/documentation/xml_format.md#interpolation)
```
Comment (bsekachev) [inline cvat/apps/documentation/user_guide.md]: ``cvat/apps/documentation/xml_format.md#interpolation``

File: cvat/apps/documentation/user_guide.md [hunk 7]
```diff
@@ -403,18 +440,31 @@ Usage examples:
...
+    is highlighted if a task has the interpolation mode
+    - [CVAT XML 1.1 for images](https://github.com/opencv/cvat/blob/develop/cvat/apps/documentation/xml_format.md#annotation)
```
Comment (bsekachev) [inline cvat/apps/documentation/user_guide.md]: ``cvat/apps/documentation/xml_format.md#annotation``

File: cvat/apps/documentation/user_guide.md [hunk 8]
```diff
@@ -603,7 +653,7 @@ Button assignment:
 - ``RunReID Merge`` — The ReID application uses deep learning model to perform
   an automatic merging of bounding boxes between neighbor frames. You can use
   "Merge" and "Split" functionality to edit automatically generated annotation
-  [read more](/cvat/apps/reid).
+  [read more](https://github.com/opencv/cvat/tree/develop/cvat/apps/reid).
```
Comment (bsekachev) [inline cvat/apps/documentation/user_guide.md]: ``cvat/apps/reid``

File: cvat/apps/documentation/user_guide.md [hunk 9]
```diff
@@ -920,6 +972,32 @@ You can add/delete points after finishing.
 
 ![](static/documentation/images/image063.jpg)
 
+### Linear interpolation with one point
+
+You can create linear interpolation for points to annotate a moving object:
```
Comment (bsekachev) [inline cvat/apps/documentation/user_guide.md]: I would prefer ``use`` instead of ``create``

File: cvat/apps/documentation/user_guide.md [hunk 10]
```diff
@@ -935,7 +1013,56 @@ in/out (when scrolling the mouse wheel) and move (when clicking the mouse
...
+
+1.  Enter model name, and select model file using "Select files" button. To annotate a task with a custom model
+    you need to prepare 4 files:
+    - ``Model config`` (*.xml) - a text file with network configuration. 
+    - ``Model weights`` (*.bin) - a binary file with trained weights. 
+    - ``Label map`` (*.json) - a simple json file with label_map dictionary like an object with 
+    string values for label numbers.
+    - ``Interpretation script`` (*.py) - a file used to convert net output layer to a predefined structure 
+    which can be processed by CVAT. 
+    More about creating model files can be found [here](https://github.com/opencv/cvat/tree/develop/cvat/apps/auto_annotation).
```
Comment (bsekachev) [inline cvat/apps/documentation/user_guide.md]: ``cvat/apps/auto_annotation``

File: cvat/apps/documentation/user_guide.md [hunk 11]
```diff
@@ -935,7 +1013,56 @@ in/out (when scrolling the mouse wheel) and move (when clicking the mouse
...
+1.  At runtime, you can see percentage of completion. You can also cancel the auto annotation
+    process by clicking ``Cancel Auto Annotation``
+
+    ![](static/documentation/images/image121.jpg)
+
+1.  As a result, you will get an annotation with separate bounding boxes (or other shapes)
+
+    ![](static/documentation/images/gif014_DETRAC.gif)
+
+1.  Separate bounding boxes can be edited by removing false positives, adding unlabeled objects, and
```
Comment (bsekachev) [inline cvat/apps/documentation/user_guide.md]: Separate -> Separated

File: cvat/apps/documentation/user_guide.md [hunk 12]
```diff
@@ -935,7 +1013,56 @@ in/out (when scrolling the mouse wheel) and move (when clicking the mouse
...
+    merged into tracks using ``Merge Shape``
```
Comment (bsekachev) [inline cvat/apps/documentation/user_guide.md]: merged -> merging

File: cvat/apps/documentation/user_guide.md [hunk 13]
```diff
@@ -935,7 +1013,56 @@ in/out (when scrolling the mouse wheel) and move (when clicking the mouse
...
+
+    ![](static/documentation/images/gif015_DETRAC.gif)
+
+Learn more about auto annotations [here](https://github.com/opencv/cvat/tree/develop/cvat/apps/auto_annotation).
```
Comment (bsekachev) [inline cvat/apps/documentation/user_guide.md]: This link have been already mentioned above twice. I am not sure it is needed again.

**Comment by bsekachev** [comment]
@TOsmanov 
Oh, something wrong with links. l am going to check it on my own.

File: cvat/apps/documentation/user_guide.md [hunk 14]
```diff
@@ -169,18 +172,39 @@ Go to the [Django administration panel](http://localhost:8080/admin). There you
...
+      is highlighted if a task has the annotation mode 
+      - [PASCAL VOC ZIP 1.0](http://host.robots.ox.ac.uk/pascal/VOC/)
+      - [YOLO ZIP 1.0](https://pjreddie.com/darknet/yolo/)
+      - [COCO JSON 1.0](http://cocodataset.org/#format-data)
+      - ``MASK ZIP 1.0`` — archive contains a mask of each frame in the png format and a text file
+      with the value of each color
+      - [TFRecord ZIP 1.0](https://www.tensorflow.org/tutorials/load_data/tf_records)
+    - ``Upload Annotation`` — upload an annotation file to the task. Several formats are available:
+      - [CVAT XML 1.1](cvat/apps/documentation/xml_format.md)
+      - [PASCAL VOC ZIP 1.0](http://host.robots.ox.ac.uk/pascal/VOC/)
```
Comment (HagegeR) [inline cvat/apps/documentation/user_guide.md]: repetitions in those lines

Comment (bsekachev) [reply cvat/apps/documentation/user_guide.md]: @HagegeR 
Why do you think so?

**Comment by bsekachev** [comment]
@TOsmanov 
Links have been fixed

**Comment by TOsmanov** [comment]
@bsekachev 
Thank you!

File: cvat/apps/documentation/user_guide.md (hunk 14 above)
Comment (HagegeR) [reply cvat/apps/documentation/user_guide.md]: line 197 and 205 are the same.
196 and 204
198 and 206
201 and 207
also 192 and 194 are very similar to 203 although there is some added value

**Pull request merged**